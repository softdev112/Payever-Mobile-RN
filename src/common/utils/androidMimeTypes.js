const mimeTypes = {
  mp3:     'audio/mpeg',
  mpga:    'audio/mpeg',
  m4a:     'audio/mp4',
  wav:     'audio/x-wav',
  amr:     'audio/amr',
  awb:     'audio/amr-wb',
  wma:     'audio/x-ms-wma',
  ogg:     'audio/ogg, application/ogg',
  aac:     'audio/aac, audio/aac-adts',
  mka:     'audio/x-matroska',
  mid:     'audio/midi',
  midi:    'audio/midi',
  xmf:     'audio/midi',
  rtttl:   'audio/midi',
  smf:     'audio/sp-midi',
  imy:     'audio/imelody',
  rtx:     'audio/midi',
  ota:     'audio/midi',
  mxmf:    'audio/midi',
  mpeg:    'video/mpeg, video/mp2p',
  mpg:     'video/mpeg, video/mp2p',
  mp4:     'video/mp4',
  m4v:     'video/mp4',
  '3gp':   'video/3gpp',
  '3gpp':  'video/3gpp',
  '3G2':   'video/3gpp2',
  '3GPP2': 'video/3gpp2',
  mkv:     'video/x-matroska',
  webm:    'video/webm',
  ts:      'video/mp2ts',
  avi:     'video/avi',
  wmv:     'video/x-ms-wmv',
  asf:     'video/x-ms-asf',
  jpg:     'image/jpeg',
  jpeg:    'image/jpeg',
  gif:     'image/gif',
  png:     'image/png',
  bmp:     'image/x-ms-bmp',
  wbmp:    'image/vnd.wap.wbmp',
  webp:    'image/webp',
  m3u:     'audio/x-mpegurl, application/x-mpegurl',
  pls:     'audio/x-scpls',
  wpl:     'application/vnd.ms-wpl',
  m3u8:    'application/vnd.apple.mpegurl, audio/mpegurl, audio/x-mpegurl',
  fl:      'application/x-android-drm-fl',
  txt:     'text/plain',
  htm:     'text/html',
  html:    'text/html',
  pdf:     'application/pdf',
  doc:     'application/msword',
  xls:     'application/vnd.ms-excel',
  ppt:     'application/mspowerpoint',
  flac:    'audio/flac',
  zip:     'application/zip',
  rar:     'application/zip',
};

function getTypeByExt(ext: string) {
  if (!ext || ext === '') return '*/*';

  let fileExt = ext;
  if (ext.startsWith('.')) {
    fileExt = ext.substr(1).toLowerCase();
  }

  return mimeTypes[fileExt] ? mimeTypes[fileExt] : '*/*';
}

export default {
  getTypeByExt,
};