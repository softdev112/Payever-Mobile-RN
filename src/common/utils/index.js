/**
 * @providesModule utils
 */

export androidMimeTypes from './androidMimeTypes';
export apiHelper from './apiHelper';
export format from './format';
export log from 'react-native-logging';
export pushNotificationsHelper from './pushNotificationsHelper';
export screenParams from './screenParams';
export soundHelper from './soundHelper';
export deepLinksHelper from './deepLinksHelper';
export networkHelper from './networkHelper';
export cacheHelper from './cacheHelper';