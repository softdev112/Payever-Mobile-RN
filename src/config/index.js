import generatedConfig from './_autogenerated';

export class Config {
  siteUrl: string;
  api: {
    baseUrl: string;
    clientId: string;
    clientSecret: string;
  };
  debug: {
    consoleLevel: boolean;
    logApiCall: boolean;
    logMobx: boolean;
    logSLevel: boolean;
    logSUrl: string;
    logWampCall: boolean;
  };

  constructor(data) {
    data.debug = data.debug || {};
    Object.assign(this, data);
  }
}

export default new Config(generatedConfig);