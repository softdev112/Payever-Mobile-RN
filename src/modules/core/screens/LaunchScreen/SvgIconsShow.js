import { Component } from 'react';
import { ART, Animated, Easing } from 'react-native';
import { View, StyleSheet } from 'ui';

import svgIcons from './icons';

const {
  Surface,
  Shape,
} = ART;

const ICON_SHOW_TIME = 450;
const ICON_CHANGE_DURATION = 700;
const START_PAYEVER_ICON_DURATION = 3000;

const DEFAULT_PATH_X = 52;
const DEFAULT_PATH_Y = 40;
const DEFAULT_SCALE = 2;

export default class SvgIconsShow extends Component {
  props: {
    style: Object;
  };

  state: {
    animValue: Object;
    currentIcon: number;
    isAnimStoped: boolean;
  };

  constructor(props) {
    super(props);

    this.startAnimation = ::this.startAnimation;

    this.state = {
      animValue: new Animated.Value(1),
      currentIcon: 0,
      isAnimStoped: false,
    };
  }

  componentDidMount() {
    setTimeout(this.startAnimation, START_PAYEVER_ICON_DURATION);
  }

  componentWillUnmount() {
    const { animValue } = this.state;

    this.setState({ isAnimStoped: true });
    animValue.stopAnimation();
  }

  startAnimation() {
    const { animValue, currentIcon, isAnimStoped } = this.state;

    Animated.timing(animValue, {
      toValue: 0,
      duration: ICON_CHANGE_DURATION,
      easing: Easing.quad,
      delay: ICON_SHOW_TIME,
    }).start(() => {
      if (isAnimStoped) return;

      const nextIcon = currentIcon === svgIcons.length - 1
        ? 0 : currentIcon + 1;
      this.setState({ currentIcon: nextIcon });

      Animated.timing(animValue, {
        toValue: 1,
        duration: ICON_CHANGE_DURATION,
        easing: Easing.quad,
      }).start(this.startAnimation);
    });
  }

  renderCurrentIcon() {
    const { currentIcon } = this.state;

    return svgIcons[currentIcon].map((iconPath, index) => {
      return (
        <Shape
          key={index}
          scale={iconPath.scale || DEFAULT_SCALE}
          d={iconPath.path}
          fill={iconPath.fill}
          stroke={iconPath.stroke}
          x={iconPath.x || DEFAULT_PATH_X}
          y={iconPath.y || DEFAULT_PATH_Y}
        />
      );
    });
  }

  render() {
    const { style } = this.props;
    const { animValue } = this.state;

    return (
      <View style={[styles.container, style]}>
        <Animated.View style={{ opacity: animValue }}>
          <Surface width={250} height={250}>
            {this.renderCurrentIcon()}
          </Surface>
        </Animated.View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});