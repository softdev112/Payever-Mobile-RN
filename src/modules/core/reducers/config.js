import { Map } from 'immutable';
import config from '../../../config';

const initialState = Map(config);

/**
 * Share config through store
 */
export default function configReducer(state = initialState) {
  return state;
}