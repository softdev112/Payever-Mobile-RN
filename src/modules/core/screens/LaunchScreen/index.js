import { Component } from 'react';
import { Linking, TouchableOpacity, AsyncStorage } from 'react-native';
import { observer, inject } from 'mobx-react/native';
import { Icon, View, Text, StyleSheet } from 'ui';
import { Navigator } from 'react-native-navigation';
import * as Animatable from 'react-native-animatable';
import { log } from 'utils';

import SvgIconsShow from './SvgIconsShow';
import type { Config } from '../../../../config';

const IS_AGREED_TAG = '@DE_PAYEVER:IS_AGREED';

const termsUrl = 'https://payever.de/en/about/terms-of-service/';
const policyUrl = 'https://getpayever.com/about/privacy/';

@inject('config')
@observer
export default class LaunchScreen extends Component {
  static navigatorStyle = { navBarHidden: true };

  props: {
    navigator: Navigator;
    config: Config;
  };

  state: {
    isAgreed: boolean;
    isSwitchToAgreed: boolean;
  };

  constructor(props) {
    super(props);

    this.state = {
      isAgreed: false,
      isSwitchToAgreed: false,
    };
  }

  async componentWillMount() {
    try {
      const isAgreed = await AsyncStorage.getItem(IS_AGREED_TAG);
      if (isAgreed !== null) {
        this.setState({ isAgreed: JSON.parse(isAgreed) });
      }
    } catch (error) {
      log.error(error);
    }
  }

  async onAgreedPress() {
    try {
      await AsyncStorage.setItem(IS_AGREED_TAG, JSON.stringify(true));
    } catch (error) {
      log.error(error);
    }

    this.setState({ isSwitchToAgreed: true });
  }

  onCreateAccountPress() {
    const { config, navigator } = this.props;
    navigator.push({
      screen: 'core.WebView',
      passProps: { url: config.siteUrl + '/register' },
    });
  }

  onSignInPress() {
    this.props.navigator.push({
      screen: 'auth.Login',
      animated: true,
    });
  }

  onTermsPress() {
    Linking.openURL(termsUrl).catch(log.error);
  }

  onPolicyPress() {
    Linking.openURL(policyUrl).catch(log.error);
  }

  render() {
    const { isAgreed, isSwitchToAgreed } = this.state;

    return (
      <View style={styles.container}>
        <SvgIconsShow style={styles.animContainer} />
        <View style={styles.infoContainer}>
          <Text style={styles.welcomeText}>Welcome to Payever</Text>
          { isAgreed ? (
            <Animatable.View
              style={styles.signInSignUpCont}
              animation={isSwitchToAgreed ? 'zoomIn' : ''}
              duration={400}
              delay={400}
              easing="ease-in-out-cubic"
              onAnimationEnd={() => this.setState({ isSwitchToAgreed: false })}
            >
              <TouchableOpacity onPress={::this.onSignInPress}>
                <View style={styles.authBtn}>
                  <Text style={styles.authBtnText}>Sign In</Text>
                  <Icon
                    style={styles.authBtnText}
                    source="icon-arrow-right-ios-16"
                  />
                </View>
              </TouchableOpacity>
              <View style={styles.divider} />
              <TouchableOpacity onPress={::this.onCreateAccountPress}>
                <View style={styles.authBtn}>
                  <Text style={styles.authBtnText}>Create Account</Text>
                  <Icon
                    style={styles.authBtnText}
                    source="icon-arrow-right-ios-16"
                  />
                </View>
              </TouchableOpacity>
            </Animatable.View>
          ) : (
            <Animatable.View
              style={styles.termsAgreedCont}
              animation={isSwitchToAgreed ? 'zoomOut' : ''}
              duration={400}
              easing="ease-in-out-cubic"
              onAnimationEnd={() => this.setState({ isAgreed: true })}
            >
              <Text style={styles.termsAndPolicyText}>
                {'Tap "Agree & Continue" to accept the Payever'}
              </Text>
              <Text>
                <Text
                  style={styles.termsAndPolicyLinks}
                  onPress={::this.onTermsPress}
                >
                  Terms of Service
                </Text>
                <Text style={styles.termsAndPolicyText}>{' and '}</Text>
                <Text
                  style={styles.termsAndPolicyLinks}
                  onPress={::this.onPolicyPress}
                >
                  Privacy Policy
                </Text>
              </Text>
              <Text
                style={styles.agreeBtn}
                onPress={::this.onAgreedPress}
              >
                Agree & Continue
              </Text>
            </Animatable.View>
          )}
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },

  animContainer: {
    flexBasis: 40,
    flex: 60,
    alignItems: 'center',
    justifyContent: 'center',
  },

  infoContainer: {
    flex: 40,
  },

  termsAgreedCont: {
    flex: 1,
    alignItems: 'center',
  },

  welcomeText: {
    fontSize: '3.7rem',
    fontWeight: '400',
    alignSelf: 'center',
    '@media ios': {
      fontFamily: 'Helvetica Neue',
    },
    '@media android': {
      fontFamily: 'Roboto',
    },
  },

  termsAndPolicyText: {
    marginTop: 20,
    fontSize: '1.7rem',
    fontWeight: '400',
    textAlign: 'center',
    '@media ios': {
      fontFamily: 'Helvetica Neue',
    },
    '@media android': {
      fontFamily: 'Roboto',
    },
  },

  termsAndPolicyLinks: {
    fontSize: '1.7rem',
    fontWeight: '400',
    color: '$pe_color_blue',
    textAlign: 'center',
    '@media ios': {
      fontFamily: 'Helvetica Neue',
    },
    '@media android': {
      fontFamily: 'Roboto',
    },
  },

  agreeBtn: {
    marginTop: '7%',
    fontSize: '2.6rem',
    fontWeight: '400',
    color: '$pe_color_blue',
  },

  signInSignUpCont: {
    flex: 1,
    justifyContent: 'center',
    paddingHorizontal: 35,
  },

  authBtn: {
    height: 50,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },

  authBtnText: {
    fontSize: '2.2rem',
    fontWeight: '400',
    color: '$pe_color_blue',

  },

  divider: {
    height: 1,
    backgroundColor: '$pe_color_light_gray_1',
  },
});