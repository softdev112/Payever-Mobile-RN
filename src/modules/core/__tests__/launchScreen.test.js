/* eslint-disable max-len */
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { deepLinksHelper } from 'utils';
import LaunchScreen from '../screens/LaunchScreen';
import Store from '../../../store';
import config from '../../../config';

jest.mock('../screens/LaunchScreen/SvgIconsShow', () => 'SvgIconsShow');

const navigator = {
  setOnNavigatorEvent: jest.fn(),
  resetTo: jest.fn(),
  push: jest.fn(),
};

describe('modules/core/screens/LaunchScreen', () => {
  let store;
  let profiles;

  beforeAll(() => {
    deepLinksHelper.processDeepLink = jest.fn();
  });

  beforeEach(() => {
    store = new Store(config);
    profiles = store.profiles;
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('LaunchScreen should render correctly and should NOT called processDeepLink if store.ui.deepLink not set', () => {
    const tree = renderer.create(
      <LaunchScreen
        profiles={profiles}
        ui={store.ui}
        navigator={navigator}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();

    // componentDidMount with ui.deepLink = ''
    expect(deepLinksHelper.processDeepLink).not.toHaveBeenCalled();
  });

  it('LaunchScreen should render correctly and should call processDeepLink if store.ui.deepLink = url', () => {
    store.ui.deepLink = 'https://mein.payever.de';

    const tree = renderer.create(
      <LaunchScreen
        profiles={profiles}
        ui={store.ui}
        navigator={navigator}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();

    // componentDidMount with ui.deepLink = url
    expect(deepLinksHelper.processDeepLink).toHaveBeenCalled();
    expect(deepLinksHelper.processDeepLink).toHaveBeenCalledWith(
      'https://mein.payever.de',
      expect.anything(),
      expect.objectContaining(navigator)
    );
  });

  it('LaunchScreen/onSignInPress() test press to signIn button should call navigator.push', () => {
    const wrapper = shallow((
      <LaunchScreen
        profiles={profiles}
        ui={store.ui}
        navigator={navigator}
      />
    ));

    wrapper.dive()
      .find('TouchableOpacity')
      .at(0)
      .props()
      .onPress();

    expect(navigator.push).toHaveBeenCalled();
    expect(navigator.push).toHaveBeenCalledWith(
      { animated: true, screen: 'auth.Login' }
    );
  });

  it('LaunchScreen/onCreateAccountPress() test press to signIn button should call navigator.push', () => {
    const wrapper = shallow((
      <LaunchScreen
        profiles={profiles}
        ui={store.ui}
        navigator={navigator}
      />
    ));

    wrapper.dive()
      .find('TouchableOpacity')
      .at(1)
      .props()
      .onPress();

    expect(navigator.push).toHaveBeenCalled();
    expect(navigator.push).toHaveBeenCalledWith(
      { animated: true, screen: 'auth.Register' }
    );
  });
});