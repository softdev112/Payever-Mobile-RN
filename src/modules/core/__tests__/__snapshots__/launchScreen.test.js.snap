// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modules/core/screens/LaunchScreen LaunchScreen should render correctly with dummy data and should NOT called processDeepLink if store.ui.deepLink not set 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <SvgIconsShow
    style={
      Object {
        "alignItems": "center",
        "flex": 60,
        "flexBasis": 40,
        "justifyContent": "center",
      }
    }
  />
  <View
    style={
      Object {
        "flex": 40,
      }
    }
  >
    <View
      style={
        Object {
          "paddingHorizontal": 16,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#3d3d3d",
              "fontFamily": "Helvetica Neue",
              "fontWeight": "200",
            },
            Object {
              "alignSelf": "center",
              "fontFamily": "Helvetica Neue",
              "fontSize": 43.4,
              "fontWeight": "300",
            },
          ]
        }
      >
        Welcome to payever
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#3d3d3d",
              "fontFamily": "Helvetica Neue",
              "fontWeight": "200",
            },
            Object {
              "color": "#7d7d7d",
              "fontFamily": "Helvetica Neue",
              "fontSize": 23.8,
              "fontWeight": "400",
              "marginTop": 20,
              "textAlign": "center",
            },
          ]
        }
      >
        Start, run and grow your business or get personal offers
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-end",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#f0eff5",
            "height": 1,
            "marginLeft": 20,
          }
        }
      />
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "paddingLeft": 20,
              "paddingRight": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#3d3d3d",
                  "fontFamily": "Helvetica Neue",
                  "fontWeight": "200",
                },
                Object {
                  "color": "#0084ff",
                  "fontSize": 16,
                  "fontWeight": "400",
                },
              ]
            }
          >
            Sign In
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            source="icon-arrow-right-ios-16"
            style={
              Array [
                Object {
                  "fontFamily": "payeverIcons",
                  "fontSize": 16,
                },
                Array [
                  Object {
                    "fontFamily": "payeverIcons",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#e1e1e1",
                    "fontSize": 16,
                    "fontWeight": "300",
                  },
                ],
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#f0eff5",
            "height": 1,
            "marginLeft": 20,
          }
        }
      />
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "paddingLeft": 20,
              "paddingRight": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#3d3d3d",
                  "fontFamily": "Helvetica Neue",
                  "fontWeight": "200",
                },
                Object {
                  "color": "#0084ff",
                  "fontSize": 16,
                  "fontWeight": "400",
                },
              ]
            }
          >
            Create Account
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            source="icon-arrow-right-ios-16"
            style={
              Array [
                Object {
                  "fontFamily": "payeverIcons",
                  "fontSize": 16,
                },
                Array [
                  Object {
                    "fontFamily": "payeverIcons",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#e1e1e1",
                    "fontSize": 16,
                    "fontWeight": "300",
                  },
                ],
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`modules/core/screens/LaunchScreen LaunchScreen should render correctly with dummy data and should call processDeepLink if store.ui.deepLink = url 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <SvgIconsShow
    style={
      Object {
        "alignItems": "center",
        "flex": 60,
        "flexBasis": 40,
        "justifyContent": "center",
      }
    }
  />
  <View
    style={
      Object {
        "flex": 40,
      }
    }
  >
    <View
      style={
        Object {
          "paddingHorizontal": 16,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#3d3d3d",
              "fontFamily": "Helvetica Neue",
              "fontWeight": "200",
            },
            Object {
              "alignSelf": "center",
              "fontFamily": "Helvetica Neue",
              "fontSize": 43.4,
              "fontWeight": "300",
            },
          ]
        }
      >
        Welcome to payever
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#3d3d3d",
              "fontFamily": "Helvetica Neue",
              "fontWeight": "200",
            },
            Object {
              "color": "#7d7d7d",
              "fontFamily": "Helvetica Neue",
              "fontSize": 23.8,
              "fontWeight": "400",
              "marginTop": 20,
              "textAlign": "center",
            },
          ]
        }
      >
        Start, run and grow your business or get personal offers
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-end",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#f0eff5",
            "height": 1,
            "marginLeft": 20,
          }
        }
      />
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "paddingLeft": 20,
              "paddingRight": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#3d3d3d",
                  "fontFamily": "Helvetica Neue",
                  "fontWeight": "200",
                },
                Object {
                  "color": "#0084ff",
                  "fontSize": 16,
                  "fontWeight": "400",
                },
              ]
            }
          >
            Sign In
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            source="icon-arrow-right-ios-16"
            style={
              Array [
                Object {
                  "fontFamily": "payeverIcons",
                  "fontSize": 16,
                },
                Array [
                  Object {
                    "fontFamily": "payeverIcons",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#e1e1e1",
                    "fontSize": 16,
                    "fontWeight": "300",
                  },
                ],
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#f0eff5",
            "height": 1,
            "marginLeft": 20,
          }
        }
      />
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "paddingLeft": 20,
              "paddingRight": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#3d3d3d",
                  "fontFamily": "Helvetica Neue",
                  "fontWeight": "200",
                },
                Object {
                  "color": "#0084ff",
                  "fontSize": 16,
                  "fontWeight": "400",
                },
              ]
            }
          >
            Create Account
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            source="icon-arrow-right-ios-16"
            style={
              Array [
                Object {
                  "fontFamily": "payeverIcons",
                  "fontSize": 16,
                },
                Array [
                  Object {
                    "fontFamily": "payeverIcons",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#e1e1e1",
                    "fontSize": 16,
                    "fontWeight": "300",
                  },
                ],
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
